dcrMetric.py

```diff
@@ -0,0 +1,301 @@
+        hour_angle: `float`
+            The observed hour angle of the observation.
+        
+        Returns
+        -------
+        visit_measure: `float`
+            A one dimensional parameterization of airmass and hour angle.
+        """
+        visit_measure = (airmass - 1.)*np.sign(hour_angle)
+        return visit_measure
+    
+    def parameterizeTable(self, visits=None):
+        """Calculate the visit measures for all observations in the database.
+        
+        Parameters
+        ----------
+        visits: `list`, optional
+            The visit identifiers of the desired observations.
+            If `None`, all observations in the database are used.
+        
+        Returns
+        -------
+        visit_measures : `list` of `float`
+            The one-dimensional parameterization of airmass and hour angle for all visits in the database.
+        """
+        visit_measures = []
+        if visits is None:
+            visits = self.table.index
+        for visit in visits:
+            airmass = self.table['airmass'][visit]
+            hour_angle = self.table['hour_angle'][visit]
+            measure = self.parameterizeVisit(airmass, hour_angle)
+            if measure >= self.range[0] and measure <= self.range[1]:
+                visit_measures.append(measure)
+        return visit_measures
+
+    def addVisit(self, visit, airmass, hour_angle):
+        """Add a new visit to the database.
+        
+        Parameters
+        ----------
+        visit: `float`
+            The visit identifier of the observation to be added.
+        airmass: `float`
+            The observed airmass of the observation.
+        hour_angle: `float`
+            The observed hour angle of the observation.
+        """
+        if visit in self.table.index:
+            raise KeyError(f"Visit {visit} is already present in the table, and cannot be added. Skipping it.")
+        self.table.loc[visit] = [airmass, hour_angle]
+        self.kde = None
+        
+    def addVisitInfo(self, visitInfo):
+        """Add a new visit to the database using the VisitInfo of an exposure.
+        
+        Parameters
+        ----------
+        visitInfo: `lsst.afw.image.VisitInfo`
+            The VisitInfo metadata of an exposure.
+        """
+        visit = visitInfo.getExposureId()//100  # This is Decam-specific
+        hour_angle = visitInfo.getBoresightHourAngle().asRadians()
+        airmass = visitInfo.getBoresightAirmass()
+        self.addVisit(visit, airmass, hour_angle)
+    
+    def delVisit(self, visit):
+        """Remove a visit from the database.
+        
+        Parameters
+        ----------
+        visit: `float`
+            The visit identifier of the observation to be removed.
+        """
+        try:
+            airmass = self.table['airmass'][visit]
+            hour_angle = self.table['hour_angle'][visit]
+            self.table.drop(visit, inplace=True)
```

Comment:
Do all three of these lines need to be in the try block, or can one or more of them happen before it or in the else block? It's not immediately clear which one you're catching an exception on. (But if it's any/all, it's fine to leave it as is.)